src/gui/torrentcontentwidget.cpp

```diff
@@ -71,8 +72,11 @@ namespace
 TorrentContentWidget::TorrentContentWidget(QWidget *parent)
     : QTreeView(parent)
 {
-    setDragEnabled(true);
+    const auto *preferences = Preferences::instance();
```

Comment:
It seems that you have chosen the wrong approach. `TorrentContentWidget` should not directly deal with `Preferences`, but the parent component should do and set the appropriate properties of `TorrentContentWidget` instance. This will allow you to handle it differently in the main window and the Add new torrent dialog (which should have content dragging unconditionally disabled, as mentioned earlier).

Reply:
The changes I've proposed are quite trivial to implement. They don't change the essence of the feature, but they will make it complete and correct from a code design perspective.